apm-agent-plugins/apm-es-restclient-plugin/apm-es-restclient-plugin-common/src/main/java/co/elastic/apm/agent/esrestclient/EndpointResolutionHelper.java

```diff
@@ -0,0 +1,536 @@
+        routesMap.put("es/ccr.stats", new String[]{"/_ccr/stats"});
+        routesMap.put("es/indices.forcemerge", new String[]{"/_forcemerge", "/{index}/_forcemerge"});
+        routesMap.put("es/indices.delete_template", new String[]{"/_template/{name}"});
+        routesMap.put("es/sql.delete_async", new String[]{"/_sql/async/delete/{id}"});
+        routesMap.put("es/security.update_api_key", new String[]{"/_security/api_key/{id}"});
+        routesMap.put("es/security.create_service_token", new String[]{"/_security/service/{namespace}/{service}/credential/token/{name}", "/_security/service/{namespace}/{service}/credential/token"});
+        routesMap.put("es/license.get_trial_status", new String[]{"/_license/trial_status"});
+        routesMap.put("es/searchable_snapshots.mount", new String[]{"/_snapshot/{repository}/{snapshot}/_mount"});
+        routesMap.put("es/security.grant_api_key", new String[]{"/_security/api_key/grant"});
+        routesMap.put("es/ilm.retry", new String[]{"/{index}/_ilm/retry"});
+        routesMap.put("es/ml.reset_job", new String[]{"/_ml/anomaly_detectors/{job_id}/_reset"});
+        routesMap.put("es/ml.close_job", new String[]{"/_ml/anomaly_detectors/{job_id}/_close"});
+        routesMap.put("es/ml.explain_data_frame_analytics", new String[]{"/_ml/data_frame/analytics/_explain", "/_ml/data_frame/analytics/{id}/_explain"});
+        routesMap.put("es/security.clear_cached_service_tokens", new String[]{"/_security/service/{namespace}/{service}/credential/token/{name}/_clear_cache"});
+        routesMap.put("es/search_mvt", new String[]{"/{index}/_mvt/{field}/{zoom}/{x}/{y}"});
+    }
+
+    protected static EndpointResolutionHelper get() {
+        if (INSTANCE == null) {
+            INSTANCE = new EndpointResolutionHelper();
+        }
+        return INSTANCE;
+    }
+
+    protected void enrichSpanWithRouteInformation(Span<?> span, String method, String endpointId, String urlPath) {
+        String[] availableRoutes = routesMap.get(endpointId);
+        if (availableRoutes == null || availableRoutes.length == 0) {
+            return;
+        }
+
+        span.withOtelAttribute("db.operation", (endpointId.startsWith("es/") && endpointId.length() > 3) ? endpointId.substring(3) : endpointId);
+        if (availableRoutes.length == 1) {
+            enrichSpan(span, method, availableRoutes[0], urlPath);
+        } else {
+            int i = 0;
+            boolean enriched = false;
+            while (i < availableRoutes.length && !enriched) {
+                enriched = enrichSpan(span, method, availableRoutes[i], urlPath);
+                i++;
+            }
+        }
+    }
+
+    private Matcher matchUrl(String route, String urlPath) {
+        if (!regexPatternMap.containsKey(route)) {
+            StringBuilder regexStr = new StringBuilder();
+            regexStr.append('^');
+            regexStr.append(route.replace("{", "(?<").replace("}", ">[^/]+)"));
+            regexStr.append('$');
```

Comment:
No need for the `^` and `$` in the regex if you use `Matcher.matches()` instead of `Matcher.find()` in `enrichSpan`.